test/hotspot/jtreg/runtime/jni/TestPrimitiveArrayCriticalWithBadParam/TestPrimitiveArrayCriticalWithBadParam.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2021, Red Hat, Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/**
+ * @test TestPrimitiveArrayCriticalWithBadParam
+ * @summary -Xcheck:jni should catch wrong parameter passed to GetPrimitiveArrayCritical
+ * @library /test/lib
+ * @run main/othervm/native TestPrimitiveArrayCriticalWithBadParam
```

Comment:
othervm is not needed, but you will need to explicitly set the library path for the launched VM.